### PR TITLE
Decouple UTXO fetching

### PR DESCRIPTION
**Pull request by Shourya742**
Closes: #143

### CODE REVIEW

File: src/wallet/api.rs [hunk 1]
```diff
@@ -637,17 +639,32 @@ impl Wallet {
...
+        let all_utxos: Vec<ListUnspentResultEntry> = self
+            .store
+            .list_unspent
+            .lock()
+            .unwrap()
+            .iter()
+            .map(|e| match e {
+                ListUnspentInfo::ListUnspentResultEntry(v) => v.clone(),
+            })
+            .collect();
```
Comment (mojoX911) [inline src/wallet/api.rs:667]: While I liked the lock/unlock utxo feature from core, but that itself is another dependency. And we should be able to filter utxos ourselves. In theory can remove this rpc dep too.

What happens if we only have one call `get_all_utxos`? The other APIs can (or already does) filter for different kind of utxos.

File: src/wallet/rpc.rs [hunk 2]
```diff
@@ -148,11 +148,47 @@ impl Wallet {
...
+            .rpc
+            .list_unspent(Some(0), Some(9999999), None, None, None)?;
+        self.store.list_unspent.lock().unwrap().clear();
+        all_utxos.into_iter().for_each(|list_unspent_result_entry| {
+            let list_unspent = ListUnspentInfo::ListUnspentResultEntry(list_unspent_result_entry);
+            self.store.list_unspent.lock().unwrap().push(list_unspent);
+        });
+        Ok(())
+    }
+
```
Comment (mojoX911) [inline src/wallet/rpc.rs:191]: Same thing. We should be able to just sync all utxos and filter at later layers.

File: src/wallet/storage.rs [hunk 3]
```diff
@@ -16,8 +17,33 @@ use super::{error::WalletError, fidelity::FidelityBond};
 
 use super::swapcoin::{IncomingSwapCoin, OutgoingSwapCoin};
 
+#[derive(Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub enum ListUnspentInfo {
+    ListUnspentResultEntry(ListUnspentResultEntry),
```
Comment (mojoX911) [inline src/wallet/storage.rs:22]: Possible to reduce `ListUnspentResultEntry` to only information we need to store? No need to just copy the whole core-rpc struct.

File: src/wallet/storage.rs [hunk 4]
```diff
@@ -37,10 +63,12 @@ pub struct WalletStore {
...
     /// Map for all the fidelity bond information. (index, (Bond, script_pubkey, is_spent)).
     pub(super) fidelity_bond: HashMap<u32, (FidelityBond, ScriptBuf, bool)>,
-    //TODO: Add last synced height and Wallet birthday.
+    /// last synced height
     pub(super) last_synced_height: Option<u64>,
-
+    /// Wallet birthday
     pub(super) wallet_birthday: Option<u64>,
+    /// list of unspent transaction output
+    pub(super) list_unspent: Mutex<Vec<ListUnspentInfo>>,
```
Comment (mojoX911) [inline src/wallet/storage.rs:71]: A RwLock could be more efficient than Mutex as we don't need to block at all read locks. RwLock allows multiple reads, but only one write lock.

File: src/wallet/storage.rs [hunk 5]
```diff
@@ -100,31 +129,3 @@ impl WalletStore {
...
-
-    use bitcoind::tempfile::tempdir;
-
-    #[test]
-    fn test_write_and_read_wallet_to_disk() {
-        let temp_dir = tempdir().unwrap();
-        let file_path = temp_dir.path().join("test_wallet.cbor");
-        let mnemonic = Mnemonic::generate(12).unwrap().to_string();
-
-        let original_wallet_store = WalletStore::init(
```
Comment (mojoX911) [inline src/wallet/storage.rs:115]: Why remove this test?

**Review by mojoX911 — COMMENTED** [review]
Ack. Looking good. Few comments.

**Comment by Shourya742** [comment]
Closing in response to last dev call discussion.

**Pull request closed without merging**